### ISSUE TITLE
[sdk] Error 400 authenticating via Google IAP

### ISSUE THREAD

**Issue by votti** [kubeflow/pipelines#7474]
### Environment

*  KFP version:
?
*  KFP SDK version: 
1.8.11
*  All dependencies version:
kfp                      1.8.11
kfp-pipeline-spec        0.1.14
kfp-server-api           1.8.1

### Steps to reproduce

I followed the steps described in the documentation to connect to Kubeflow Pipelines in a full Kubeflow deployment with authentication through IAP:
https://www.kubeflow.org/docs/distributions/gke/pipelines/authentication-sdk/#connecting-to-kubeflow-pipelines-in-a-full-kubeflow-deployment

Following the Oauth link from the SDK:
_3. To connect to Kubeflow Pipelines public endpoint, initiate SDK client like the following:_
```
import kfp
client = kfp.Client(host='https://<KF_NAME>.endpoints.<PROJECT>.cloud.goog/pipeline',
    client_id='<AAAAAAAAAAAAAAAAAAAAAA>.apps.googleusercontent.com',
    other_client_id='<BBBBBBBBBBBBBBBBBBB>.apps.googleusercontent.com',
    other_client_secret='<CCCCCCCCCCCCCCCCCCCC>')
```
I get an Error 400: 
![grafik](https://user-images.githubusercontent.com/3743016/160302422-7959f30b-ad43-45b9-9f6b-6611700494e8.png)

### Expected result
I should get an authentication screen and then a code.

### Materials and Reference
This error seems to a deprecation of the copy-paste workflow for oauth authentification at the google side, as discussed here:
https://stackoverflow.com/questions/71318804/google-oauth-2-0-failing-with-error-400-invalid-request-for-some-client-id-but/71327990#71327990

I also implemented the workaround discussed here: https://stackoverflow.com/a/71331562
essentially "Replace urn:ietf:wg:oauth:2.0:oob with http://localhost:1/"

It is not sufficient to just change the `redirect_uri` parameter in url returned in the  vanilla 1.8.11. A change of this parameter in the source code is required as well.
This seems to work in my quick POC patch of KFP v1.8.11: https://github.com/votti/pipelines/tree/fix_redirect_uri_1.8.11 
Installable via: `pip install "git+https://github.com/votti/pipelines@fix_redirect_uri_1.7#subdirectory=sdk/python"`

--
Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by ji-yaqi** [comment]
Hi @votti, feel free to send a PR and we will take a look. If it involves an interface change, we would need a design. Thank you!

**Comment by pablofiumara** [comment]
@ji-yaqi Have you all had time to review this related PR? https://github.com/kubeflow/pipelines/pull/7526

**Comment by gkcalat** [comment]
Hi @votti.

We have implemented a workaround for OOB in #8262. It should be included in the next SDK releases for v2 and v1.8.

You can also build it from **master** branch if you want to test earlier:
```bash
mkdir testkfp && cd testkfp
git clone git@github.com:kubeflow/pipelines
cd pipelines/sdk/python
rm -r ./dist/
python3 setup.py bdist_wheel
```

This will create a wheel in **pipelines/sdk/python/dist** folder. You can create a new virtual environment to install the wheel:
```bash
cd ./dist
python3 -m venv testenv
. ./testenv/bin/activate
pip install ./kfp-2.0.0b4-py3-none-any.whl    # filename could vary 
```

To test it, run thin in python (replace parameters below):
```python
import kfp
client = kfp.Client(host='https://<KF_NAME>.endpoints.<PROJECT>.cloud.goog/pipeline',
    client_id='<AAAAAAAAAAAAAAAAAAAAAA>.apps.googleusercontent.com',
    other_client_id='<BBBBBBBBBBBBBBBBBBB>.apps.googleusercontent.com',
    other_client_secret='<CCCCCCCCCCCCCCCCCCCC>')
client.list_pipelines()
```

Let me know if it works.
